etc/common.sh

```diff
@@ -228,7 +247,9 @@ EOF
     yum install -y \
       kubelet${KUBE_VERSION} \
       kubeadm${KUBE_VERSION} \
-      kubectl${KUBE_VERSION}
+      kubectl${KUBE_VERSION} \
+      kubernetes-cni${KUBE_CNI_VERSION}
```

Comment:
Why do we need the CNI to use GPUs?

Reply:
This is a dependency for kubernetes packages, not for GPUs.
I was not able to install kubelet, kubeadm and kubekubectl  without installl kubernetes-cni.

Reply:
True, but this is dependency which is automatically brought in by the package manager (otherwise, it would have never have worked):
```
============================================================================================================================================================================================================
 Package                                                  Arch                                     Version                                               Repository                                    Size
============================================================================================================================================================================================================
Installing:
 kubeadm                                                  x86_64                                   1.15.1-0                                              kubernetes                                   8.9 M
 kubectl                                                  x86_64                                   1.15.1-0                                              kubernetes                                   9.5 M
 kubelet                                                  x86_64                                   1.15.1-0                                              kubernetes                                    22 M
Installing for dependencies:
 conntrack-tools                                          x86_64                                   1.4.4-4.el7                                           base                                         186 k
 cri-tools                                                x86_64                                   1.13.0-0                                              kubernetes                                   5.1 M
 ebtables                                                 x86_64                                   2.0.10-16.el7                                         base                                         123 k
 ethtool                                                  x86_64                                   2:4.8-9.el7                                           base                                         127 k
 iproute                                                  x86_64                                   4.11.0-14.el7_6.2                                     updates                                      764 k
 iptables                                                 x86_64                                   1.4.21-28.el7                                         base                                         433 k
 kubernetes-cni                                           x86_64                                   0.7.5-0                                               kubernetes                                    10 M
 libmnl                                                   x86_64                                   1.0.3-7.el7                                           base                                          23 k
 libnetfilter_conntrack                                   x86_64                                   1.0.6-1.el7_3                                         base                                          55 k
 libnetfilter_cthelper                                    x86_64                                   1.0.0-9.el7                                           base                                          18 k
 libnetfilter_cttimeout                                   x86_64                                   1.0.0-6.el7                                           base                                          18 k
 libnetfilter_queue                                       x86_64                                   1.0.2-2.el7_2                                         base                                          23 k
 libnfnetlink                                             x86_64                                   1.0.1-4.el7                                           base                                          26 k
 socat                                                    x86_64                                   1.7.3.2-2.el7                                         base                                         290 k
 tcp_wrappers-libs                                        x86_64                                   7.6-77.el7                                            base                                          66 k

Transaction Summary
============================================================================================================================================================================================================
Install  3 Packages (+15 Dependent packages)
```

Could you please check this again?
I am not too much in favor of adding this to the list as it brings in the `KUBE_CNI_VERSION`, which will need to be kept in sync with `KUBE_VERSION`.

Thanks!

Reply:
Hi @ebocchi 
using a clean installation of a virtual machine in OpenStack with the image 
CC7 - x86_64 [2018-12-03] I am getting this without  kubernetes-cni



Resolving Dependencies
--> Running transaction check
---> Package kubeadm.x86_64 0:1.10.11-0 will be installed
--> Processing Dependency: kubernetes-cni >= 0.6.0 for package: kubeadm-1.10.11-0.x86_64
--> Processing Dependency: cri-tools >= 1.11.0 for package: kubeadm-1.10.11-0.x86_64
---> Package kubectl.x86_64 0:1.10.11-0 will be installed
---> Package kubelet.x86_64 0:1.10.11-0 will be installed
--> Processing Dependency: kubernetes-cni = 0.6.0 for package: kubelet-1.10.11-0.x86_64
--> Processing Dependency: socat for package: kubelet-1.10.11-0.x86_64
--> Running transaction check
---> Package cri-tools.x86_64 0:1.13.0-0 will be installed
---> Package kubelet.x86_64 0:1.10.11-0 will be installed
--> Processing Dependency: kubernetes-cni = 0.6.0 for package: kubelet-1.10.11-0.x86_64
---> Package kubernetes-cni.x86_64 0:0.7.5-0 will be installed
---> Package socat.x86_64 0:1.7.3.2-2.el7 will be installed
--> Finished Dependency Resolution
Beginning Kernel Module Plugin
Finished Kernel Module Plugin
Error: Package: kubelet-1.10.11-0.x86_64 (kubernetes)
           Requires: kubernetes-cni = 0.6.0
           Available: kubernetes-cni-0.3.0.1-0.07a8a2.x86_64 (kubernetes)
               kubernetes-cni = 0.3.0.1-0.07a8a2
           Available: kubernetes-cni-0.5.1-0.x86_64 (kubernetes)
               kubernetes-cni = 0.5.1-0
           Available: kubernetes-cni-0.5.1-1.x86_64 (kubernetes)
               kubernetes-cni = 0.5.1-1
           Available: kubernetes-cni-0.6.0-0.x86_64 (kubernetes)
               kubernetes-cni = 0.6.0-0
           Installing: kubernetes-cni-0.7.5-0.x86_64 (kubernetes)
               kubernetes-cni = 0.7.5-0
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodiges

Some suggestions?
Thanks!

Reply:
Is the k8s repo correctly configured?
If you are getting these packages from somewhere else, it might be that the dependencies are not correctly resolved.

Here is what I have:
```
=====================================================================================================
 Package                         Arch            Version                   Repository           Size
=====================================================================================================
Installing:
 kubeadm                         x86_64          1.15.1-0                  kubernetes          8.9 M
Installing for dependencies:
 conntrack-tools                 x86_64          1.4.4-4.el7               base                186 k
 cri-tools                       x86_64          1.13.0-0                  kubernetes          5.1 M
 kubectl                         x86_64          1.15.1-0                  kubernetes          9.5 M
 kubelet                         x86_64          1.15.1-0                  kubernetes           22 M
 kubernetes-cni                  x86_64          0.7.5-0                   kubernetes           10 M
[...]
```

And the k8s repo:
```
[root@edocker ~]# cat /etc/yum.repos.d/kubernetes.repo 
[kubernetes]
name=Kubernetes
baseurl=https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64
enabled=1
gpgcheck=1
repo_gpgcheck=1
gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg
https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
```

Let me know

Reply:
PS
This is another VM in OpenStack with the same OS you mentioned

Reply:
yes,
[root@test-install-kube kuboxed]# cat /etc/yum.repos.d/kubernetes.repo 
[kubernetes]
name=Kubernetes
baseurl=https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64
enabled=1
gpgcheck=1
repo_gpgcheck=1
gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg
        https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg

But I can see that the version that you are trying to install is different
kubelet                         x86_64          1.15.1-0  
I have 1.10.11-0 not the 1.15.1-0.

Do you think is better to install the version  1.15.1-0   ?

Reply:
I do not think the exact version matters.
`kubernetes-cni` should be always brought is as a dependency

Reply:
Hi @ebocchi 
Yes, it is the version, lets check this 

yum install -y kubelet-1.10.11-0 kubeadm-1.10.11-0 kubectl-1.10.11-0

this is the command execute in InitNode.sh without CNI and you will get 
Error: Package: kubelet-1.10.11-0.x86_64 (kubernetes)
           Requires: kubernetes-cni = 0.6.0
           Available: kubernetes-cni-0.3.0.1-0.07a8a2.x86_64 (kubernetes)
               kubernetes-cni = 0.3.0.1-0.07a8a2
           Available: kubernetes-cni-0.5.1-0.x86_64 (kubernetes)
               kubernetes-cni = 0.5.1-0
           Available: kubernetes-cni-0.5.1-1.x86_64 (kubernetes)
               kubernetes-cni = 0.5.1-1
           Available: kubernetes-cni-0.6.0-0.x86_64 (kubernetes)
               kubernetes-cni = 0.6.0-0
           Installing: kubernetes-cni-0.7.5-0.x86_64 (kubernetes)
               kubernetes-cni = 0.7.5-0

but If you try 1.15 you dont need to put cni
yum install -y kubelet-1.15.1-0 kubeadm-1.15.1-0 kubectl-1.15.1-0
it will work..

for the version 1.10.11 I need to set  kubernetes-cni-0.6.0
yum install -y kubelet-1.10.11-0 kubeadm-1.10.11-0 kubectl-1.10.11-0 kubernetes-cni-0.6.0
it works.

Reply:
Interesting...
Is there any reason to pin to version 1.10?
Otherwise, I would just jump to 1.15 and the problem is fixed

Reply:
I try with 1.15.1,
But there are some errors, running InitNode.sh that is not related to GPU 

1)
Configuring cgroup driver for kubelet service...
cp: cannot stat ‘/etc/systemd/system/kubelet.service.d/10-kubeadm.conf’: No such file or directory
sed: can't read /etc/systemd/system/kubelet.service.d/10-kubeadm.conf: No such file or directory

the directory /etc/systemd/system/kubelet.service.d/ do not exist for this version and I think that the default driver is not cgroup, it is systemd (but I am not sure yet)

2)
Initializing the cluster...
[init] Using Kubernetes version: v1.15.2
[preflight] Running pre-flight checks
        [WARNING IsDockerSystemdCheck]: detected "cgroupfs" as the Docker cgroup driver. The recommended driver is "systemd". Please follow the guide at https://kubernetes.io/docs/setup/cri/
error execution phase preflight: [preflight] Some fatal errors occurred:
        [ERROR NumCPU]: the number of available CPUs 1 is less than the required 2
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`
cp: cannot stat ‘/etc/kubernetes/admin.conf’: No such file or directory
chown: cannot access ‘/root/.kube/config’: No such file or directory

I found this post for the version 1.13 the problem should be for versions >=1.13
https://linuxacademy.com/community/show/33403-faild-to-run-kubeadm-init-on-centos7--error-numcpu/


I tried to install versions 1.11 and 1.12 but it fails because requires the same with the CNI package.

Some suggestions?
Should I try to move it to the version 1.15 anyway?